.github/workflows/nightly-build.yml

```diff
@@ -0,0 +1,69 @@
+name: Upload Nightly Binaries
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # run at mid night
+  workflow_dispatch:
+
+jobs:
+
+  build:
+    name: Upload Nightly Binaries
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Create tag
+        run: |
+          export tag=nightly-build/$(date +'%m-%d-%Y')/binaries.zip
+          echo $tag
+          echo "tag=$tag" >> $GITHUB_ENV
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.17
+        id: go
+
+      - name: Install zip
+        run: sudo apt install zip
+
+      - name: Install rpm
+        run: sudo apt install rpm
+
+      # Must clone a fresh repo because of how the .git folder is used in the versioning of the agent for packaging
```

Comment:
I'm not exactly sure what your trying to fix, but the way I understand it, github created this: https://github.com/actions/checkout

Reply:
It has to do with how git tags work.  When using checkout we lose these tags thus we need to clone a fresh repo. What you have posted will not work. https://github.com/actions/checkout explains by default checkout uses the event sha